torch/backends/_nnapi/prepare.py

```diff
@@ -35,7 +36,8 @@ def __init__(
     @torch.jit.export
     def init(self, args: List[torch.Tensor]):
         assert self.comp is None
-        self.out_templates = self.shape_compute_module.prepare(self.ser_model, args)
+        # self.shape_compute_prepare is a module and not a Tensor
+        self.out_templates = self.shape_compute_module.prepare(self.ser_model, args)  # type: ignore[operator]
```

Comment:
Just checking, did you try if this passes tests?
```
         self.out_mem_fmts = out_mem_fmts
-        self.out_templates = []
+        self.out_templates : List[torch.Tensor] = []
         self.comp = None
 
     @torch.jit.export
     def init(self, args: List[torch.Tensor]):
         assert self.comp is None
+        assert isinstance(self.shape_compute_module.prepare, list)
         self.out_templates = self.shape_compute_module.prepare(self.ser_model, args)
```
`reveal_type(self.shape_compute_module.prepare)` says: `Union[torch._tensor.Tensor, torch.nn.modules.module.Module]`, so using an assert seems like the better way to do type narrowing.

Reply:
I didn't but any attempt of asserting the type of `prepare` results in the following error:

```
RuntimeError:
method cannot be used as a value:
  File "/home/guilhermel/git/pytorch/torch/backends/_nnapi/prepare.py", line 39
    def init(self, args: List[torch.Tensor]):
        assert self.comp is None
        assert isinstance(self.shape_compute_module.prepare, torch.Tensor)
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        self.out_templates = self.shape_compute_module.prepare(self.ser_model, args)  # type: ignore[operator]
        self.weights = [w.contiguous() for w in self.weights]
'NnapiModule.init' is being compiled since it was called from 'NnapiModule.forward'
  File "/home/guilhermel/git/pytorch/torch/backends/_nnapi/prepare.py", line 48
    def forward(self, args: List[torch.Tensor]) -> List[torch.Tensor]:
        if self.comp is None:
            self.init(args)
                      ~~~~ <--- HERE
        comp = self.comp
        assert comp is not None
```

Reply:
I tried for a bit, there indeed doesn't seem to be a way to make mypy happy. So let's leave it as is.